enter-env.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p bashInteractive vault awscli jq openssh
+
+set -eu
+
+scratch=$(mktemp -d -t tmp.XXXXXXXXXX)
+function finish {
+  rm -rf "$scratch"
+  if [ "x${VAULT_EXIT_ACCESSOR:-}" != "x" ]; then
+    echo "--> Revoking my token ..." >&2
+    vault token revoke -self || true
+  fi
+}
+trap finish EXIT
+
+echo "--> Assuming role: packet-nix-builder-deployers" >&2
+vault_creds=$(vault token create \
+	-display-name=grahamc-packet-nix-builder \
+	-format=json \
+	-role packet-nix-builder-deployers)
+
+VAULT_EXIT_ACCESSOR=$(jq -r .auth.accessor <<<"$vault_creds")
+expiration_ts=$(($(date '+%s') + "$(jq -r .auth.lease_duration<<<"$vault_creds")"))
+export VAULT_TOKEN=$(jq -r .auth.client_token <<<"$vault_creds")
+
+echo "--> Setting variables: PACKET_AUTH_TOKEN, PACKET_PROJECT_ID, AWS_ACCESS_KEY_ID" >&2
+echo "                       AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN" >&2
+export PACKET_AUTH_TOKEN=$(vault kv get -field api_key_token packet/creds/grahamc)
+export PACKET_PROJECT_ID=$(vault kv get \
+    -field=project_id secret/buildkite/nixos-foundation/packet-project-id)
+
+aws_creds=$(vault kv get -format=json aws-personal/creds/state-packet-nix-builder)
+export AWS_ACCESS_KEY_ID=$(jq -r .data.access_key <<<"$aws_creds")
+export AWS_SECRET_ACCESS_KEY=$(jq -r .data.secret_key <<<"$aws_creds")
+export AWS_SESSION_TOKEN=$(jq -r .data.security_token <<<"$aws_creds")
+if [ -z "$AWS_SESSION_TOKEN" ] ||  [ "$AWS_SESSION_TOKEN" == "null" ]; then
+  unset AWS_SESSION_TOKEN
+fi
+
+for  i in $(seq 1 100); do
+  if aws sts get-caller-identity > /dev/null; then
```

Comment:
Does this log what it is doing? Otherwise we might want to add a statement before this. Makes it more obvious what went wrong in the failure case.

Reply:
Actually I can't seem to figure out why we need AWS credentials at all. What am I missing?

Reply:
:+1: